third_party/tensorpipe.BUILD

```diff
@@ -71,63 +71,82 @@ cc_library(
 )
 
 header_template_rule(
-    name = "tensorpipe_config_header",
+    name = "tensorpipe_cpu_config_header",
     src = "tensorpipe/config.h.in",
     out = "tensorpipe/config.h",
     substitutions = {
-        "#cmakedefine01 TENSORPIPE_HAS_SHM_TRANSPORT": "",
-        "#cmakedefine01 TENSORPIPE_HAS_CMA_CHANNEL": "",
-        "#cmakedefine01 TENSORPIPE_HAS_CUDA_IPC_CHANNEL": "",
-        "#cmakedefine01 TENSORPIPE_HAS_CUDA_GDR_CHANNEL": "",
-        "#cmakedefine01 TENSORPIPE_HAS_IBV_TRANSPORT": "",
-        "#cmakedefine01 TENSORPIPE_SUPPORTS_CUDA": "",
+        "#cmakedefine01 TENSORPIPE_HAS_SHM_TRANSPORT": "#define TENSORPIPE_HAS_SHM_TRANSPORT 1",
+        "#cmakedefine01 TENSORPIPE_HAS_IBV_TRANSPORT": "#define TENSORPIPE_HAS_IBV_TRANSPORT 1",
```

Comment:
Since we don't have CPU + IB today, would I be correct if I assume this is added for future changes in cross-device type support?

Reply:
I was actually being imprecise earlier: we do have CPU + IB today, it's just that we've optimized it for latency, and not for bandwidth (thus it's mainly useful to transfer payloads rather than tensors).